text/0000-inline-asm.md

```diff
@@ -614,6 +593,42 @@ If the `noreturn` flag is set then an `unreachable` LLVM instruction is inserted
 [llvm-clobber]: http://llvm.org/docs/LangRef.html#clobber-constraints
 [issue-65452]: https://github.com/rust-lang/rust/issues/65452
 
+## Rules for inline assembly
+[rules]: #rules
+
+- Any registers not specified as inputs will contain an undefined value on entry to the asm block.
```

Comment:
...unless they are otherwise specified by the ABI. For instance, inline assembly on x86-64 Linux should be able to use fs to access thread-local storage (unless the developer has done something unusual to make that not work).

Reply:
maybe it would be good to specify that the normally register-allocatable registers are undefined, but the other registers are defined to their standard ABI values, giving a exhaustive list somewhere, as that may be clearer.

For x86_64, the normally register-allocatable registers are:
* general purpose registers: rax, rcx, rdx, rbx, rbp, rsi, rdi, r8-r15
* SIMD registers: zmm0-31 and sub-registers
* mask registers: k0-k7
* rflags: df, of, zf, pf, sf, af
* mxcsr: pe, ue, oe, ze, de, ie
* st(0)-st(7)
* fsw
* any others I missed

Reply:
Note that the list is different than the ABI, since that only applies at function boundaries. Inside functions, registers such as x86's direction flag that are defined by the ABI can be modified, but must be reset before the next function boundary.